app/models/item.rb

```diff
@@ -30,4 +30,12 @@ def sorted_reviews(limit, order)
   def no_orders?
     item_orders.empty?
   end
+
+  def self.five_most_popular_items
+    joins(:item_orders).group(:id).select('items.name, sum(quantity) as purchase_amount').order("purchase_amount desc").limit(5)
+  end
+
+  def self.five_least_popular_items
+    joins(:item_orders).group(:id).select('items.name, sum(quantity) as purchase_amount').order("purchase_amount").limit(5)
+  end
```

Comment:
The only difference between these two methods is that one sorts ascending and the other sorts descending. I tried to make it work as just one method that takes an argument to select how it is sorted but I couldn't get that to work.  We can probably refactor.

Reply:
Since you're using a string as an argument for `.order`, I would assume you can use string interpolation, but I'd have to try it out.

Reply:
I think Michael had an interesting suggestion. We also need to test these at the model level.